apigee/adapter.go

```diff
@@ -391,27 +391,24 @@ func (h *handler) HandleQuota(ctx context.Context, inst *quotaT.Instance, args a
 func resolveClaims(log adapter.Logger, claimsIn map[string]string) map[string]interface{} {
 	var claims = map[string]interface{}{}
 	for _, k := range auth.AllValidClaims {
-		if v, ok := claims[k]; ok {
+		if v, ok := claimsIn[k]; ok {
 			claims[k] = v
 		}
 	}
 	if len(claims) > 0 {
 		return claims
 	}
 
-	var err error
 	if encoded, ok := claimsIn[encodedClaimsKey]; ok {
 		if encoded == "" {
 			return claims
 		}
-		var decoded []byte
-		decoded, err = base64.StdEncoding.DecodeString(encoded)
 
-		// hack: weird truncation issue coming from Istio, add suffix and try again
-		if err != nil && strings.HasPrefix(err.Error(), "illegal base64 data") {
-			decoded, err = base64.StdEncoding.DecodeString(encoded + "o=")
+		if len(encoded)%4 != 0 {
+			// Weird base64 bug, need to pad with =.
```

Comment:
You might also mention here what the bug is and where

Reply:
Done, I think that's what you mean.